src/main/java/org/thingsboard/trendz/generator/utils/DateTimeUtils.java

```diff
@@ -27,4 +28,15 @@ public static ZonedDateTime fromTs(Timestamp ts, ZoneId zoneId) {
     public static long toTs(ZonedDateTime zonedDateTime) {
         return zonedDateTime.toInstant().toEpochMilli();
     }
+
+    public static Pair<Long, Long> getDatesIntersection(long startDate, long endDate, long startGenerationDate, long endGenerationDate) {
+        final long latestStartDate = Math.max(startDate, startGenerationDate);
+        final long earliestEndDate = Math.min(endDate, endGenerationDate);
+
+        if (earliestEndDate < latestStartDate) {
```

Comment:
what if equal

Reply:
if both are equal, then both are returned in Pair